PsfLauncher/main.cpp

```diff
@@ -550,54 +618,84 @@ ErrorInformation CheckIfPowershellIsInstalled(bool& isPowershellInstalled) noexc
     return {};
 }
 
+ErrorInformation CheckIfShouldRun(bool shouldRunOnce, bool &canScriptRun)

```

Comment:
This should go inside the PowershellScript object. Since you are already using an object to define script running, might as well put everything about scripts in that object.

Reply:
Done.  Changed.  Thank you for the idea of returning an HRESULT.